### PR TITLE
Initial recipe for mlxtend

### PR DESCRIPTION
**Pull request by rasbt**
Adds a recipe for the [mlxtend](https://github.com/rasbt/mlxtend) Python package.

### CODE REVIEW

**Comment by souravsingh** [comment]
@rasbt You won't need ``build.sh`` for the recipe.

File: recipes/mlxtend/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    The documentation can be found at http://rasbt.github.io/mlxtend/
+
+
+    '
+  doc_url: 'http://rasbt.github.io/mlxtend/'
+  dev_url: 'https://github.com/rasbt/mlxtend'
+
+extra:
+  recipe-maintainers:
+    - Sebastian Raschka
```
Comment (souravsingh) [inline recipes/mlxtend/meta.yaml]: Use Github username here

File: recipes/mlxtend/build.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+$PYTHON setup.py install --single-version-externally-managed --record record.txt
```
Comment (souravsingh) [inline recipes/mlxtend/build.sh]: You won't require this since this is taken care of at the recipe itself.

File: recipes/mlxtend/meta.yaml (hunk 1 above)
Comment (rasbt) [inline recipes/mlxtend/meta.yaml]: done!

File: recipes/mlxtend/build.sh (hunk 2 above)
Comment (rasbt) [inline recipes/mlxtend/build.sh]: thanks, just removed it

File: recipes/mlxtend/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  fn: '{{ name }}-{{ version }}.tar.gz'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: python setup.py install  --single-version-externally-managed --record=record.txt
+
+requirements:
+  build:
```
Comment (synapticarbors) [inline recipes/mlxtend/meta.yaml]: I *think* only `python` and `setuptools` are required to build the package.

File: recipes/mlxtend/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    - mlxtend.text
+    - mlxtend.utils
+
+about:
+  home: https://github.com/rasbt/mlxtend
+  license: BSD v3
+  license_family: BSD
+  license_file: LICENSE-BSD3.txt
+  summary: Machine Learning Library Extensions
+  description: '
```
Comment (synapticarbors) [inline recipes/mlxtend/meta.yaml]: There are a bunch of extra blank lines in the `description`.  You can use either a `|` or a `>` at the start for literal or folded style respectively:

http://symfony.com/doc/current/components/yaml/yaml_format.html

File: recipes/mlxtend/meta.yaml (hunk 3 above)
Comment (rasbt) [inline recipes/mlxtend/meta.yaml]: Essentially, you are right. Different functions within mlxtend require different packages as dependencies though (i.e., numpy, scipy, scikit-learn, pandas, and matplotlib). However, having those packages it is not required for installing mlxtend. I will remove these dependencies then.

Comment (synapticarbors) [inline recipes/mlxtend/meta.yaml]: It's more of what the `setup.py` actually needs to build the package. The `--single-version-externally-managed` makes it such that all you need to provide are external libraries that are imported by running `setup.py`.

File: recipes/mlxtend/meta.yaml (hunk 4 above)
Comment (rasbt) [inline recipes/mlxtend/meta.yaml]: Thanks, updated it

File: recipes/mlxtend/meta.yaml (hunk 3 above)
Comment (rasbt) [inline recipes/mlxtend/meta.yaml]: Okay, thanks. I think I got it now :P

**Pull request merged**